src/rgw/rgw_asio_frontend.cc

```diff
@@ -221,7 +223,29 @@ class AsioFrontend {
 
 int AsioFrontend::init()
 {
-  auto ep = tcp::endpoint{tcp::v4(), static_cast<unsigned short>(env.port)};
+  std::string port_str;
+  conf->get_val("port", "80", &port_str);
+
+  std::string ip_addr;
+  unsigned short port;
+
+  if (!boost::algorithm::contains(port_str, ":")) {
+    ip_addr = "0.0.0.0";
+    port = std::stol(port_str, nullptr, 0);
+  } else {
+    std::vector<std::string> confs;
+    get_str_vec(port_str, ":", confs);
+
+    if (confs.size() != 2) {
+      lderr(ctx()) << "failed to parse port: " << port_str << dendl;
+      return -1;
+    }
+
+    ip_addr = confs[0];
+    port = std::stol(confs[1], nullptr, 0);
+  }
+
+  tcp::endpoint ep = {boost::asio::ip::address::from_string(ip_addr), port};
```

Comment:
according to the [docs for from_string()](http://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ip__address/from_string.html), it's deprecated in favor of [make_address()](http://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ip__address/make_address.html) - can you use the overload that takes `string_view` and avoid the use of vector/string? something like:
```c++
  tcp::address addr; // default to 'any'
  boost::string_view input = port_str;
  auto colon = input.find(':');
  if (colon != input.npos) {
    boost::system::error_code ec;
    addr = boost::asio::ip::make_address(input.substr(0, colon), ec);
    if (ec) {
      lderr(ctx()) << "failed to parse address '" << port_str << "': " << ec.message() << dendl;
      return -ec.value();
    }
  }
  tcp::endpoint ep = {addr, port};
```

then eventually we can extend the frontend to listen on multiple sockets